From 9babbf223fbdccce6bbe24c6448c23bbf990e399 Mon Sep 17 00:00:00 2001
From: liushuyu <liushuyu011@gmail.com>
Date: Tue, 27 Dec 2022 00:53:51 -0700
Subject: [PATCH 2/2] static: defer the load of branding CSS and ...

... run the boot function immediately when the DOM is parsed
---
 pkg/static/login.html | 9 +++++----
 pkg/static/login.js   | 2 +-
 2 files changed, 6 insertions(+), 5 deletions(-)

diff --git a/pkg/static/login.html b/pkg/static/login.html
index f6eb27c..973fb61 100644
--- a/pkg/static/login.html
+++ b/pkg/static/login.html
@@ -12,13 +12,14 @@
     if (window.NodeList && !NodeList.prototype.forEach)
         NodeList.prototype.forEach = Array.prototype.forEach;
   </script>
-  <script type="text/javascript">/*insert_translations_here*/</script>
-  <script type="text/javascript" src="cockpit/static/login.js"></script>
-  <link href="cockpit/static/login.css" type="text/css" rel="stylesheet">
-  <link href="cockpit/static/branding.css" type="text/css" rel="stylesheet">
 </head>
 
 <body class="login-pf">
+  <script async type="text/javascript">/*insert_translations_here*/</script>
+  <script defer type="text/javascript" src="cockpit/static/login.js"></script>
+  <link href="cockpit/static/login.css" type="text/css" rel="stylesheet">
+  <link href="cockpit/static/branding.css" type="text/css" rel="stylesheet">
+
   <div id="banner" class="pf-c-alert pf-m-info pf-m-inline dialog-error" aria-label="inline danger alert" hidden>
     <svg fill="currentColor" viewBox="0 0 448 512" aria-hidden="true">
       <path d="M224 512c35.32 0 63.97-28.65 63.97-64H160.03c0 35.35 28.65 64 63.97 64zm215.39-149.71c-19.32-20.76-55.47-51.99-55.47-154.29 0-77.7-54.48-139.9-127.94-155.16V32c0-17.67-14.32-32-31.98-32s-31.98 14.33-31.98 32v20.84C118.56 68.1 64.08 130.3 64.08 208c0 102.3-36.15 133.53-55.47 154.29-6 6.45-8.66 14.16-8.61 21.71.11 16.4 12.98 32 32.1 32h383.8c19.12 0 32-15.6 32.1-32 .05-7.55-2.61-15.27-8.61-21.71z" />
diff --git a/pkg/static/login.js b/pkg/static/login.js
index 363d0b6..d3bcab8 100644
--- a/pkg/static/login.js
+++ b/pkg/static/login.js
@@ -1039,5 +1039,5 @@
         login_reload(wanted);
     }
 
-    window.onload = boot;
+    boot();
 })(window.console);
-- 
2.39.0

